apps/web/pages/api/auth/verify-email.ts

```diff
@@ -26,7 +26,20 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
     return res.status(401).json({ message: "Token expired" });
   }
 
-  const user = await prisma.user.update({
+  // Find the parent user to verify
+  const user = await prisma.user.findFirst({
+    where: {
+      email: foundToken?.identifier,
+      linkedBy: null,
+    },
+    select: {
+      id: true,
+      completedOnboarding: true,
+    },
+  });
+
+  // Proceeding to update all matching users by email as verified
+  await prisma.user.updateMany({
```

Comment:
Not being able to use `prisma.user.update` makes us find the parent user to define onboarding status and just update all user rows with the same email as email verified.